Gems/Atom/RPI/Code/Source/RPI.Public/Pass/CopyPass.cpp

```diff
@@ -138,11 +294,323 @@ namespace AZ
             }
         }
 
-        void CopyPass::BuildCommandListInternal(const RHI::FrameGraphExecuteContext& context)
+        void CopyPass::BuildCommandListInternalSameDevice(const RHI::FrameGraphExecuteContext& context)
+        {
+            if (m_copyItemSameDevice.m_type != RHI::CopyItemType::Invalid)
+            {
+                context.GetCommandList()->Submit(m_copyItemSameDevice.GetDeviceCopyItem(context.GetDeviceIndex()));
+            }
+        }
+
+        void CopyPass::SetupFrameGraphDependenciesDeviceToHost(RHI::FrameGraphInterface frameGraph)
         {
-            if (m_copyItem.m_type != RHI::CopyItemType::Invalid)
+            // We need the size of the output image when copying from image to image, so we need all attachments (even the output ones)
+            // We also need it so the framegraph knows the two scopes depend on each other
+            for (const PassAttachmentBinding& attachmentBinding : m_attachmentBindings)
+            {
+                if (attachmentBinding.GetAttachment() != nullptr &&
+                    frameGraph.GetAttachmentDatabase().IsAttachmentValid(attachmentBinding.GetAttachment()->GetAttachmentId()))
+                {
+                    switch (attachmentBinding.m_unifiedScopeDesc.GetType())
+                    {
+                    case RHI::AttachmentType::Image:
+                        {
+                            frameGraph.UseAttachment(
+                                attachmentBinding.m_unifiedScopeDesc.GetAsImage(),
+                                attachmentBinding.GetAttachmentAccess(),
+                                attachmentBinding.m_scopeAttachmentUsage);
+                            break;
+                        }
+                    case RHI::AttachmentType::Buffer:
+                        {
+                            frameGraph.UseAttachment(
+                                attachmentBinding.m_unifiedScopeDesc.GetAsBuffer(),
+                                attachmentBinding.GetAttachmentAccess(),
+                                attachmentBinding.m_scopeAttachmentUsage);
+                            break;
+                        }
+                    default:
+                        AZ_Assert(false, "Error, trying to bind an attachment that is neither an image nor a buffer!");
+                        break;
+                    }
+                }
+            }
+            for (Pass* pass : m_executeAfterPasses)
             {
-                context.GetCommandList()->Submit(m_copyItem.GetDeviceCopyItem(context.GetDeviceIndex()));
+                RenderPass* renderPass = azrtti_cast<RenderPass*>(pass);
+                if (renderPass)
+                {
+                    frameGraph.ExecuteAfter(renderPass->GetScopeId());
+                }
+            }
```

Comment:
Lets remove this as no one should be using m_executeAfterPasses and if they are we should be removing it as ExecuteAfter does not imply execution dependency.